extension/src/popup/views/SignTransaction.tsx

```diff
@@ -87,13 +90,26 @@ export const SignTransaction = () => {
     ? JSON.parse(decodedTransactionInfo)
     : {};
   const {
-    tab: { title },
+    tab: { title, url },
     transaction,
   } = transactionInfo;
 
   const { _fee, _operations } = transaction;
   const publicKey = useSelector(publicKeySelector);
   const [isConfirming, setIsConfirming] = useState(false);
+  const operationTypes = _operations.map(
+    (operation: { type: string }) => operation.type,
+  );
+
+  const METRIC_OPTIONS = {
+    domain: url,
+    number_of_operations: _operations.length,
+    operationTypes,
+  };
+
+  useEffect(() => {
```

Comment:
I'm not happy with how I implemented it so suggestions welcome.

Reply:
Is this needed? "View" metrics are being fired by `popup/metrics/views.ts`

Reply:
ooh is this needed bc you need to pass the metric a `body`?

Reply:
looks like this will double report `METRIC_NAMES.viewSignTransaction` since it will happen on page load from `popup/metrics/views` and then again on `useEffect`.

Reply:
It is needed to fire the options. I can remove the one from `popup/metrics/views.ts`. It's just not pretty 😢

Reply:
I'm going to try a different approach.

Reply:
I thought of creating another slice for metrics but it isn't necessary since it isn't being used elsewhere other than for metrics. I also don't want to make an exception case by telling it to skip `/sign-transaction` in `popup/metrics/views.ts` 🤔